ega-data-api-htsget/src/main/java/eu/elixir/ega/ebi/htsget/formats/VCFDataProvider.java

```diff
@@ -0,0 +1,85 @@
+package eu.elixir.ega.ebi.htsget.formats;
+
+import eu.elixir.ega.ebi.htsget.dto.HtsgetResponseV2;
+import eu.elixir.ega.ebi.htsget.dto.HtsgetUrlV2;
+import htsjdk.samtools.seekablestream.SeekableStream;
+import htsjdk.samtools.util.BlockCompressedInputStream;
+import htsjdk.samtools.util.BlockCompressedOutputStream;
+import htsjdk.samtools.util.BlockCompressedStreamConstants;
+import htsjdk.tribble.FeatureCodecHeader;
+import htsjdk.tribble.index.Block;
+import htsjdk.tribble.index.tabix.TabixIndex;
+import htsjdk.variant.variantcontext.writer.Options;
+import htsjdk.variant.variantcontext.writer.VariantContextWriter;
+import htsjdk.variant.variantcontext.writer.VariantContextWriterBuilder;
+import htsjdk.variant.vcf.VCFCodec;
+import htsjdk.variant.vcf.VCFHeader;
+import org.apache.commons.io.input.CloseShieldInputStream;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class VCFDataProvider extends AbstractDataProvider implements DataProvider {
+    VCFHeader header;
+
+    public VCFDataProvider(SeekableStream dataStream) throws IOException {
+        super(dataStream);
+    }
+
+    @Override
+    public boolean supportsFileType(String filename) {
+        return filename.toLowerCase().endsWith(".vcf.gz");
```

Comment:
The other DataProviders had a different pattern, e.g.:
```
return filename.toLowerCase().endsWith(".bam") || filename.toLowerCase().endsWith(".bam.cip");
```
I know nothing about VCF files, so I don't know whether this `endsWith(".vcf.gz")` should be 
```
return filename.toLowerCase().endsWith(".vcf") || filename.toLowerCase().endsWith(".vcf.gz") || filename.toLowerCase().endsWith(".vcf.cip") || filename.toLowerCase().endsWith(".vcf.gz.cip");
```
or not.

Reply:
After talking with Jorge, we realised that it is probably not possible to slice uncompressed VCF because Tabix does not support making indexes for uncompressed VCF. So it is correct that we do not return true for that extension here, we do not want the VCFDataProvider to try and handle uncompressed VCF, we just want to tell the user that it is not supported.